src/mito2/src/read/seq_scan.rs

```diff
@@ -152,6 +153,10 @@ impl SeqScan {
         } else {
             self.build_reader().await?
         };
+        reader = Box::new(NullExpandReader::new(
+            reader,
+            self.mapper.metadata().clone(),
+        ));
```

Comment:
I wonder if we should expand nulls in the memtable iter. Otherwise, it might cause potential issues as we assume batches from the same iter/reader have the same schema (now some of them have null data type).

Reply:
Memtable might be too early.

We should special handle `NullVector` and treat it as a typed vector.

BTW, `Batch` doesn't contain concrete data type. Bring it with column data might alleviate this problem

Reply:
We have data type in the array/vector. If we expand nulls in the memtable iter or implement a NullExpandIterator, we can ensure Batches have the same schema. We can further delay expansion if we have better support for typed `NullVector`.

cc @v0y4g3r @fengjiachun

Reply:
data type in vector and array are not enough. It prohibits us from using `NullVector` to express an all-null array.

I'm making a `TypedNullVector`, which should be the same thing you want. But I'm considering changing it to something like `IdenticalVector` for a wilder use scenario. How does this look to you?

Reply:
Maybe we can support attaching a logical type to the `NullVector`.